File: gc-commons/gc-base-common/src/main/java/com/gc/common/base/imports/EnableGlobalException.java
package com.gc.common.base.imports;

import com.gc.common.base.exception.handler.GlobalExceptionHandler;
import org.springframework.context.annotation.Import;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * @author jackson
 * 2020/2/15 9:14 下午
 */
@Target(ElementType.TYPE)
@Retention(RetentionPolicy.RUNTIME)
@Import(GlobalExceptionHandler.class)
public @interface EnableGlobalException {
}

File: smart-commons/smart-common/src/main/kotlin/com/smart/common/import/EnableGlobalException.kt
package com.smart.common.import

import com.smart.common.spring.GlobalExceptionHandler
import org.springframework.context.annotation.Import

/**
 * 全局异常处理
 * @author ming
 * 2019/9/10 上午9:47
 */
@Import(GlobalExceptionHandler ::class)
annotation class EnableGlobalException {
}